pyproject.toml

```diff
@@ -1,5 +1,99 @@
-# https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html
+[build-system]
+requires = ["setuptools", "wheel"]
+
+[tool.black]
+line-length = 88
+target_version = ["py38"]
+exclude = '\.ipynb'
 
-# isort compatibility.
 [tool.isort]
-profile = "black"
\ No newline at end of file
+profile = "black"
+include_trailing_comma = true
+combine_as_imports = true
+
+[tool.mypy]
+# Basic mypy configuration.
+check_untyped_defs = true
+explicit_package_bases = true
+follow_imports = "silent"
+ignore_missing_imports = true
+namespace_packages = true
+pretty = true
+python_version = "3.8"
+show_error_codes = true
+strict = true
+plugins = "numpy.typing.mypy_plugin"
+
+# Lint rules.
+allow_redefinition = false
+disallow_any_generics = true
+disallow_incomplete_defs = true
+disallow_subclassing_any = true
+disallow_untyped_calls = true
+disallow_untyped_decorators = true
+disallow_untyped_defs = true
+no_implicit_reexport = true
+no_implicit_optional = true
+strict_equality = true
+strict_optional = true
+warn_redundant_casts = true
+warn_return_any = true
+warn_unused_ignores = false
+warn_unused_configs = true
+warn_unreachable = true
+
+[tool.pylint.master]
+jobs = 4
+good-names = ["i", "j", "k", "v", "e", "df", "f", "fp", "X", "x", "y", "db"]
```

Comment:
I agree.  Pylint will complain if you ever use any very short variable name that isn't whitelisted.  Each of these names has a specific usage in mind, but I haven't documented this.  Some are more obvious than others.  "f" is for `with open("file.txt", "r") as f:`.  Others are more obscure. `X x and y` are all for ML modeling. `db` instead of database felt reasonable.  What do you think we should do overall?  I'm open to change this however you see fit

Reply:
I think we leave this discussion open and merge as-is for now